app/models/endpoint.rb

```diff
@@ -3,6 +3,7 @@ class Endpoint < ApplicationRecord
 
   default_value_for :verify_ssl, OpenSSL::SSL::VERIFY_PEER
   validates :verify_ssl, :inclusion => {:in => [OpenSSL::SSL::VERIFY_NONE, OpenSSL::SSL::VERIFY_PEER]}
+  validates :port, :numericality => {:only_integer => true, :allow_nil => true, :greater_than => 0}
```

Comment:
@juliancheal @Fryguy just a small note to the "integer column already" in db: I was debugging the case when user fills port number with characters (non-numeric) and finished on http://apidock.com/rails/v4.2.1/ActiveRecord/AttributeMethods/Write/write_attribute_with_type_cast and the returned value was the same as users input, but in https://github.com/ManageIQ/manageiq/blob/master/app/controllers/ems_common.rb#L872 sets port to 0 (before this PR when user left empty port number or write non-numeric, now it sets to 0 only when non-numeric value, but the validation revoke the value  with 'not a number' error message - what's was requested to be fixed) ... it's weird that empty port value or non-numeric value is translated to the DB as 0, isn't it? could it be somehow related to the virtual column definition ?